### DISCUSSION TITLE
Could not CreateTrayIcon(): IWindowingPlatform is not registered when running ControlCatalog.NetCore sample in framebuffer

### DISCUSSION THREAD

**Discussion by prabhatkk** [AvaloniaUI/Avalonia#7828] [Q&A] [answered]
I am running Ubuntu 20.04 on HyperV. When I run the same Avalonia App from my virtual terminal using the following command line
**FRAMEBUFFER=/dev/fb0 dotnet exec ControlCatalog.NetCore.dll --fbdev**
I get this error
_System.Exception
  HResult=0x80131500
  Message=Could not CreateTrayIcon(): IWindowingPlatform is not registered.
  Source=Avalonia.Controls
  StackTrace:
   at Avalonia.Controls.Platform.PlatformManager.CreateTrayIcon() in \Avalonia\src\Avalonia.Controls\Platform\PlatformManager.cs:line 30
at Avalonia.Controls.TrayIcon..ctor() in Avalonia\src\Avalonia.Controls\TrayIcon.cs:line 44
   at ControlCatalog.App.!XamlIlPopulate(IServiceProvider , App ) in Avalonia\samples\ControlCatalog\App.xaml:line 36_

<img width="599" alt="image" src="https://user-images.githubusercontent.com/27871353/158922809-216784ad-dd61-464c-9791-5315b4fd2d4d.png">

Ami doing something wrong here or there is a bug here?

**Comment by kekekeks** [comment, marked as answer]
You have a tray icon in your App.xaml. Tray icons are not supported in framebuffer mode.

**Reply by prabhatkk** [reply]
Thanks Nikita. I have got it running now.  Any suggestion on how to interact with the sample controls. 
<img width="724" alt="image" src="https://user-images.githubusercontent.com/27871353/159102905-75cfa575-193a-46c4-870b-39cfff00bf6c.png">
I can't use mouse, keyboard or touch to interact with it. If I click on any point I end up seeing black pixels which is basically the terminal behind it. This is running on a HyperV virtual terminal.
<img width="467" alt="image" src="https://user-images.githubusercontent.com/27871353/159103051-0ea79238-4734-4a08-a454-12033704bda7.png">

**Reply by kekekeks** [reply]
fbdev currently only supports touch input since it's supposed to be used with embedded devices. Make sure that your HyperV VM has touch input enabled. `EvTouch USB Graphics Tablet` emulated by QEMU worked fine during our tests.

**Reply by prabhatkk** [reply]
The host for the HyperV doesn't have Touch input, so that wont work unfortunately. Sorry, I am a noob., so some follow up questions: 
I was hoping to get my virtual environment in Ubuntu setup with QEMU, but could not find online on how to setup EvTouch emulation with QEMU. Could you share the command line you used to run QEMU on linux. My assumption is you ran something like qemu-system-i386 -usbdevice tablet  (with additional options to load a linux image?). If there is an article on this I missed, pls point me to it.  Thanks Nikita.

**Reply by kekekeks** [reply]
virt-manager with libvirt+KVM should provide you with EvTouch tablet by default.

**Reply by prabhatkk** [reply]
I tried that, EvTouch is enabled by default, still can't get to interact with using mouse or Keyboard. I tried connecting an external USB Mouse, a [Bamboo Wacom Pen Tablet](https://cdn.wacom.com/f/manuals/en/bamboo-users-manual.pdf) via USB, tried redirecting the USB to my VM towards no avail. Here is my QEMU/KVM Boot options. Do you reckon that changing any of the I/O configs will make it work ?

[KVM Boot Options.txt](https://github.com/AvaloniaUI/Avalonia/files/8368010/KVM.Boot.Options.txt)

**Reply by prabhatkk** [reply]
@kekekeks In my pursuit to get this working I hooked my laptop with a touch screen. The touch works great on Ubuntu X Windows Apps, but not on Avalonia Sample app running on FB.  Here is an output from **lsusb** command. Any suggestions or ideas on how to make this work?
Bus 001 Device 002: ID 0627:0001 Adomax Technology Co., Ltd QEMU USB Tablet
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID
1d6b:0001 Linux Foundation 1.1 root hub
Bus 003 Device 004: ID 2575:0402 Weida Hi-Tech
Coo1TouchR System
Bus 003 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
